lib/orderbook/OrderBook.ts

```diff
@@ -117,10 +117,9 @@ class OrderBook extends EventEmitter {
       });
     }
     if (remainingOrder) {
-      this.broadcastOrder(remainingOrder);
+      this.broadcastOrder(remainingOrder as orders.StampedOwnOrder);
```

Comment:
Instead of casting `remainingOrder` here, I think it makes more sense to change the `MatchingResult` type to use `StampedOwnOrder`. We can also change the `match` function to only accept own orders, since we shouldn't be matching peer orders.